File: Templates/Project Templates/macOS/Application Extension/Smart Card Token Extension.xctemplate/TokenDriver.m
//___FILEHEADER___

#import "Token.h"

@implementation TokenDriver

- (TKSmartCardToken *)tokenDriver:(TKSmartCardTokenDriver *)driver createTokenForSmartCard:(TKSmartCard *)smartCard AID:(NSData *)AID error:(NSError **)error {
    return [[Token alloc] initWithSmartCard:smartCard AID:AID tokenDriver:self error:error];
}

@end

File: Templates/Project Templates/macOS/Application Extension/Smart Card Token Extension.xctemplate/TokenDriver.swift
//___FILEHEADER___

import CryptoTokenKit

class TokenDriver: TKSmartCardTokenDriver, TKSmartCardTokenDriverDelegate {

    func tokenDriver(_ driver: TKSmartCardTokenDriver, createTokenFor smartCard: TKSmartCard, aid AID: Data?) throws -> TKSmartCardToken {
        return try Token(smartCard: smartCard, aid: AID, tokenDriver: self)
    }

}
